src/optimise.jl

```diff
@@ -1,17 +1,29 @@
-
+﻿
+export HeapedArray,SortedArray
 """
+    minimise(f, X, str=SortedArray, tol=1e-3) 
+    or
+    minimise(f, X, str=HeapedArray, tol=1e-3)
+    or 
     minimise(f, X, tol=1e-3)
 
-Find the global minimum of the function `f` over the `Interval` or `IntervalBox` `X` using the Moore-Skelboe algorithm.
+Find the global minimum of the function `f` over the `Interval` or `IntervalBox` `X` using the Moore-Skelboe algorithm. By specifing the way in which vector element are kept arranged which is in heaped array or in sorted array. If you not specify any particular strategy to keep vector elements arranged then by default heaped array is used.  
 
 For higher-dimensional functions ``f:\\mathbb{R}^n \\to \\mathbb{R}``, `f` must take a single vector argument.
 
 Returns an interval containing the global minimum, and a list of boxes that contain the minimisers.
 """
-function minimise(f, X::T, tol=1e-3) where {T}
+
+
+defstra=HeapedArray
+function minimise(f, X::T ;str::Strategy=defstra ,tol=1e-3 ) where {T}
 
     # list of boxes with corresponding lower bound, ordered by increasing lower bound:
-    working = SortedVector([(X, ∞)], x->x[2])
+    if str==SortedArray
```

Comment:
You should never need to do this kind of `if` checking in Julia. Note that the code that you call for `working` has *exactly* the same structure in both cases: you construct a new data structure with the same arguments.

Thus the idea is rather that the variable `structure` should be either `SortedVector` or `Heap` -- i.e. the type of the object to construct -- and then you just call
``
`
working = structure( [(X, ∞)], x->x[2]) )
```

Reply:
No in both cases `working` can be object of different data structure's `SortedVector` or `heap`.
I tried with different approach, although approach you mentioned can also be an option, do I really need to change it.

Reply:
Yes, this is the point -- in that way you make the code *generic*: if someone comes along with a different data type, *no* modification to the code will then be required.